packages/beacon-node/src/api/impl/validator/utils.ts

```diff
@@ -41,3 +43,30 @@ export function getPubkeysForIndices(
 
   return pubkeys;
 }
+
+export function selectBlockProductionSource({
+  builderSelection,
+  engineBlockValue,
+  builderBlockValue,
+  builderBoostFactor,
+}: {
+  builderSelection: routes.validator.BuilderSelection;
+  engineBlockValue: bigint;
+  builderBlockValue: bigint;
+  builderBoostFactor: bigint;
+}): ProducedBlockSource {
+  if (builderSelection === routes.validator.BuilderSelection.ExecutionOnly) {
```

Comment:
a switch on builder selection is better because its more typesafe (without using default, because we plan to add `executionalways`)